compiler/kernel/expr.go

```diff
@@ -46,7 +59,7 @@ import (
 // TBD: string values and net.IP address do not need to be copied because they
 // are allocated by go libraries and temporary buffers are not used.  This will
 // change down the road when we implement no-allocation string and IP conversion.
-func (b *Builder) compileExpr(e dag.Expr) (expr.Evaluator, error) {
+func (b *Builder) compileExpr(ectx *exprContext, e dag.Expr) (expr.Evaluator, error) {
```

Comment:
I think we talked about putting this inside Builder as another possible approach, and I think I prefer that now that we can see how much noise we're creating by plumbing it through as a method parameter.